framework/src/fviks/FVDiffusionInterface.C

```diff
@@ -0,0 +1,63 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVDiffusionInterface.h"
+#include "FVUtils.h"
+
+registerMooseObject("MooseApp", FVDiffusionInterface);
+
+InputParameters
+FVDiffusionInterface::validParams()
+{
+  InputParameters params = FVInterfaceKernel::validParams();
+  params.addClassDescription("Computes the residual for diffusion operator across an interface for "
+                             "the finite volume method.");
+  params.addRequiredParam<MaterialPropertyName>("coeff1",
+                                                "The diffusion coefficient on the 1st subdomain");
+  params.addRequiredParam<MaterialPropertyName>("coeff2",
+                                                "The diffusion coefficient on the 2nd subdomain");
+  params.set<unsigned short>("ghost_layers") = 2;
+  return params;
+}
+
+FVDiffusionInterface::FVDiffusionInterface(const InputParameters & params)
+  : FVInterfaceKernel(params),
+    _coeff1_elem(getADMaterialProperty<Real>("coeff1")),
+    _coeff2_elem(getADMaterialProperty<Real>("coeff2")),
+    _coeff1_neighbor(getNeighborADMaterialProperty<Real>("coeff1")),
+    _coeff2_neighbor(getNeighborADMaterialProperty<Real>("coeff2"))
+{
+}
+
+ADReal
+FVDiffusionInterface::computeQpResidual()
+{
+  const auto & coef_elem = elemIsOne() ? _coeff1_elem : _coeff2_elem;
+  const auto & coef_neighbor = elemIsOne() ? _coeff2_neighbor : _coeff1_neighbor;
+
+  // Form a finite difference gradient across the interface
+  Point du = _face_info->elem().centroid() - _face_info->neighborPtr()->centroid();
+  du /= (du * du);
```

Comment:
I find `du` confusing. I've grokked it now but it took me a few minutes. Could you name this variable differently? E.g. something like `dxmag` and then do division by that value instead of multiplication?

Reply:
I dont think I can divide by Point